mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapperTest.java

```diff
@@ -69,22 +82,363 @@
+            testee.save(inbox);
+
+            doReturn(Mono.error(new RuntimeException("mock exception")))
+                .when(mailboxDAO)
+                .delete(inboxId);
+
+            doQuietly(() -> testee.delete(inbox));
+
+            SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+                softly.assertThatCode(() -> testee.findMailboxById(inboxId))
+                    .doesNotThrowAnyException();
+                softly.assertThatCode(() -> testee.findMailboxByPath(inboxPath))
+                    .doesNotThrowAnyException();
+                softly.assertThat(testee.findMailboxWithPathLike(inboxSearchQuery))
+                    .containsOnly(inbox);
+            }));
+        }
+
+        @Disabled("JAMES-3056 both mailboxes of the same user have 'INBOX' name")
+        @Test
+        void missedMigrationShouldNotLeadToGhostMailbox() throws Exception {
+            testee.save(inbox);
+            // simulate mailbox old data has not been migrated to v2
+            mailboxPathDAO.save(inboxPath, inboxId).block();
+            mailboxPathV2DAO.delete(inboxPath).block();
+
+            // on current v2 generation, save a new mailbox with the exactly name
+            // => two mailboxes with same name but different ids
+            CassandraId newId = CassandraId.timeBased();
+            Mailbox mailboxHasSameNameWithInbox = new Mailbox(inboxPath, UID_VALIDITY, newId);
+            testee.save(mailboxHasSameNameWithInbox);
+
+            assertThat(testee.findMailboxById(newId).getName())
+                .isNotEqualTo(testee.findMailboxById(inboxId).getName());
+        }
+
+        @Disabled("JAMES-3056 org.apache.james.mailbox.exception.MailboxNotFoundException: 'mailboxId' can not be found")
+        @Test
+        void saveAfterPreviousFailedSaveShouldCreateAMailbox() {
+            when(mailboxDAO.save(inbox))
+                .thenReturn(Mono.error(new RuntimeException("mock exception")))
+                .thenCallRealMethod();
+
+            doQuietly(() -> testee.save(inbox));
+            doQuietly(() -> testee.save(inbox));
+
+            SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+                softly.assertThat(testee.findMailboxById(inboxId))
+                    .isEqualTo(inbox);
+                softly.assertThat(testee.findMailboxByPath(inboxPath))
+                    .isEqualTo(inbox);
+                softly.assertThat(testee.findMailboxWithPathLike(inboxSearchQuery))
+                    .containsOnly(inbox);
+            }));
+        }
+
+        @Test
+        void saveAfterPreviousDeleteOnFailedSaveShouldCreateAMailbox() {
+            when(mailboxDAO.save(inbox))
+                .thenReturn(Mono.error(new RuntimeException("mock exception")))
+                .thenCallRealMethod();
+
+            doQuietly(() -> testee.save(inbox));
+            doQuietly(() -> testee.delete(inbox));
+            doQuietly(() -> testee.save(inbox));
```

Comment:
Can we write the exact same test at the manager level?

I'm pretty sure it fails, given that `StoreMailboxManager::deleteMailbox` does a findMailboxByPath before the actual delete...

Wich means we can not just tell the admin `just nuke the half created mailboxes with the webadmin user mailboxes DELETE route` as a way to leverage the inconsistency.

Reply:
I would like to write it at another PR, this's starting to big for me. and I don't think setting up mailbox manager test with some spied DAOs can be done quickly.

Reply:
FYI your test is in #3113 `createMailboxAfterDeletingShouldCreateTheMailbox...()`